kernel/src/fs/configfs/tsm.rs

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{boxed::Box, string::ToString, sync::Arc};
+use core::fmt::Debug;
+
+use aster_systree::{
+    inherit_sys_branch_node, inherit_sys_leaf_node, BranchNodeFields, Error, NormalNodeFields,
+    Result, SysAttrSetBuilder, SysObj, SysPerms, SysStr,
+};
+use aster_util::printer::VmPrinter;
+use inherit_methods_macro::inherit_methods;
+use ostd::{
+    mm::{FallibleVmRead, FallibleVmWrite, VmReader, VmWriter},
+    sync::RwLock,
+};
+use spin::Once;
+
+use crate::device::tdxguest::tdx_get_quote;
+
+const TSM_INBLOB_MAX: usize = 64;
+
+#[derive(Debug)]
+struct Tsm {
```

Comment:
Move `tsm.rs` out of configfs to separate mechanism from policies. Both `SysFs` and `ConfigFs` are designed to be mechanisms: they allow a kernel subsystem or module to expose their information or functionality to the user space via file system interfaces. Such kernel subsystems or modules are "policies"; they are users of `SysFs` or `ConfigFs`, not part of the FS implementation.

I suggest adding a new `security.rs` file under `kernel/src` and moving `tsm.rs` to it.